packages/render-fidelity-tools/src/components/renderers/filament-viewer.ts

```diff
@@ -149,37 +152,40 @@ export class FilamentViewer extends LitElement {
     console.log('Scenario:', scenario.name);
     console.log('Lighting:', lightingBaseName);
 
-    // TODO : replace this with destroyAsset() when it's released
-    if (this[$currentAsset] != null) {
-      const entities = this[$currentAsset]!.getEntities();
-      const size = entities.length;
-
-      for (let i = 0; i < size; ++i) {
-        const entity = entities[i];
-        this[$scene].remove(entity);
-        this[$engine].destroyEntity(entity);
-      }
-
+    const existingAsset = this[$currentAsset];
+    if (existingAsset != null) {
+      const entities = existingAsset.getEntities();
+      this[$scene].removeEntities(entities);
+      this[$assetLoader]!.destroyAsset(existingAsset);
       this[$currentAsset] = null;
+      for (const entity of entities) {
+        entity.delete();
+      }
     }
 
     if (this[$ibl] != null) {
       this[$scene].setIndirectLight(null);
+      this[$engine].destroyTexture(this[$ibl]!.getReflectionsTexture());
```

Comment:
We can remove all these `!` by doing `const ibl = this[$ibl]` and then operating no `ibl` instead. Actually, I'd really rather just strip all the Symbols out of this file, but I'm not going to make you do that. :)

Reply:
Added a commit to clean this up.